system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -92,6 +94,44 @@ bool PeriodicMeasurementNode::SetupStart()
   return true;
 }
 
+rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+PeriodicMeasurementNode::on_activate(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_DEBUG(this->get_logger(), "on_activate");
+  auto const ret = Start();
```

Comment:
Are `Start`, `Stop`, etc. private now? I think the interface should be either the lifecycle API, or the internal API, but not both.

Reply:
The downside to making Collector::Start and Collector::Stop private would be for any other uses outside of ROS2 (e.g. ROS1).

Reply:
We need to mark Start and Stop as protected, so the PeriodicNode API can call them for the state transitions.